book/guide/7-mint-os-token.md

```diff
@@ -0,0 +1,66 @@
+# 3. Minting Functionality
+
+In order to mint, there are a couple checks that need to be done:
+
+1. Get the max amount of shares that can be minted. You can achieve this by calling `getMaxMintForVault` on the OpusPool instance:
+
+```typescript
+let amountToMint = 100n; // the amount of shares the user wants to mint, expressed in wei
+const pool = new OpusPool({ network, address: userAddress }); // userAddress is the address of the user, can be obtained from wagmi, network is the network you are on and can be Mainnet or Holesky.
+const maxMint = await pool.getMaxMintForVault(vault);
+if (maxMint < amountToMint) {
+    // the user is trying to mint more than they can
+    return;
+}
+```
+
+2. Get the health of the vault. This can be achieved by calling the `getOsTokenPositionForVault` method on the OpusPool instance. You can learn more about the health of a vault [here](https://docs.stakewise.io/guides/oseth#maintaining-a-healthy-oseth-position).
+
+```typescript
+const { health } = await pool.getHealthFactorForVault(vaultAddress);
+await pool.getOsTokenPositionForVault(vaultAddress);
+
+if (health !== VaultHealth.Healthy) {
+    // the vault is not healthy, the user cannot mint
+    return;
+}
+```
```

Comment:
It would be useful to provide link to list of all os token position health statuses as they are represented in SDK, e.g. to docs/enums/OsTokenPositionHealth page

Reply:
The link to all possible health statuses are above, in the description of point 2.

Reply:
Thanks. I was refererring to internal documentation link, but after I've looked through guide again it seems we're referring demo website mostly instead of API docs, so maybe it should be fine to refer upstream docs here.

Still, does the type in example need to be updated to be renamed from `VaultHealth` to `OsTokenPositionHealth`   ? It appears the return type of `getOsTokenPositionForVault` method still uses `OsTokenPositionHealth` for an enum, and in fact `VaultHealth` enum is not added by this PR.

Reply:
Also it seems the other pages in guide usually include imports for custom types (for example https://github.com/ChorusOne/opus-pool-sdk/blob/main/book/guide/5-transactions-history.md?plain=1#L44), so maybe it makes sense to include this enum import here too